src/yeoman-process-adapter.ts

```diff
@@ -0,0 +1,221 @@
+/*********************************************************************
+* Copyright (c) 2018 Red Hat, Inc.
+*
+* This program and the accompanying materials are made
+* available under the terms of the Eclipse Public License 2.0
+* which is available at https://www.eclipse.org/legal/epl-2.0/
+*
+* SPDX-License-Identifier: EPL-2.0
+**********************************************************************/
+
+import * as theia from '@theia/plugin';
+import { ChildProcess } from 'child_process';
+
+const path = require('path');
+const fork = require('child_process').fork;
+
+export class YeomanProcessAdapter {
+
+    private outputChannel: theia.OutputChannel;
+
+    private childProcess: ChildProcess | undefined;
+
+    constructor(
+        protected extensionPath: string
+    ) {
+        this.outputChannel = theia.window.createOutputChannel('Yeoman Generator');
+        this.run();
+    }
+
+    /**
+     * Destroys currently running process.
+     */
+    public destroy() {
+        if (this.childProcess) {
+            this.childProcess.kill();
+        }
+    }
+
+    /**
+     * Launches separate process.
+     */
+    private run() {
+        const workspaceFolders = theia.workspace.workspaceFolders;
+        if (!workspaceFolders || workspaceFolders.length === 0) {
+            theia.window.showErrorMessage('No workspace opened, Need to open a workspace first');
+            return;
+        }
+
+        this.outputChannel.show(true);
+
+        // grab first workspace
+        const currentWorkspace = workspaceFolders[0];
+
+        // start Yeoman generator in separate node instance
+        const scriptPath = this.extensionPath + '/lib/yeoman-process.js';
+        const process = path.resolve(scriptPath);
+
+        const params: any[] = [];
+        const options = {
+            cwd: currentWorkspace.uri.path,
+            stdio: ['pipe', process.stdout, process.stderr, 'ipc']
+        };
+
+        const child = fork(process, params, options);
+
+        child.on('message', this.onMessage);
+
+        child.stdout.on('data', (data: any) => {
+            console.log(`## stdout: ${data}`);
```

Comment:
why do we need to output to the console stdout or stderr  ?

Reply:
It helps us to understand the reason when something goes wrong.
Once it helped me to catch an error when running the generator.

Reply:
why it's not part of the output on the UI side then ?

IMHO we shouldn't pollute that console by default (= not when debugging)

Reply:
UI displays only the messages related to creating the extension.
The messages form stdout / stderr are printed to Theia backend console, which is not visible in Theia.
It's easy to ignore them, but should we?

Reply:
I have reworked it a bit. All the output has been printed to the UI. Looks nice.